lib/inject.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+  def my_inject(acc = self.first, &block)
+    self.shift if acc == self.first
```

Comment:
Nice try but I can break your method by calling it like so: [2,2,3,4,5 ].my_inject(2) { |m, e| m + e }. This returns 16 when it should return 18.